astropy/modeling/fitting.py

```diff
@@ -1181,81 +1297,168 @@ def __call__(self, model, x, y, z=None, weights=None,
+            max_nfev=maxiter, diff_step=np.sqrt(epsilon), xtol=acc,
+            method=self._method, bounds=bounds
+        )
+
+        # Adapted from ~scipy.optimize.minpack, see:
+        # https://github.com/scipy/scipy/blob/47bb6febaa10658c72962b9615d5d5aa2513fa3a/scipy/optimize/minpack.py#L795-L816
+        # Do Moore-Penrose inverse discarding zero singular values.
+        _, s, VT = svd(self.fit_info.jac, full_matrices=False)
+        threshold = np.finfo(float).eps * max(self.fit_info.jac.shape) * s[0]
+        s = s[s > threshold]
+        VT = VT[:s.size]
+        cov_x = np.dot(VT.T / s**2, VT)
+
+        fitter_to_model_params(model, self.fit_info.x, False)
+        if not self.fit_info.success:
+            warnings.warn("The fit may be unsuccessful; check: \n"
+                          f"    {self.fit_info.message}",
+                          AstropyUserWarning)
 
-            pars = [getattr(model, name) for name in model.param_names]
-            fixed = [par.fixed for par in pars]
-            tied = [par.tied for par in pars]
-            tied = list(np.where([par.tied is not False for par in pars],
-                                 True, tied))
-            fix_and_tie = np.logical_or(fixed, tied)
-            ind = np.logical_not(fix_and_tie)
+        return init_values, self.fit_info.x, cov_x
 
-            if not model.col_fit_deriv:
-                residues = np.asarray(full_deriv[np.nonzero(ind)]).T
-            else:
-                residues = full_deriv[np.nonzero(ind)]
 
-            return [np.ravel(_) for _ in residues]
-        else:
-            if z is None:
-                try:
-                    return np.array([np.ravel(_) for _ in np.array(weights) *
-                                     np.array(model.fit_deriv(x, *params))])
-                except ValueError:
-                    return np.array([np.ravel(_) for _ in np.array(weights) *
-                                     np.moveaxis(
-                                         np.array(model.fit_deriv(x, *params)),
-                                         -1, 0)]).transpose()
-            else:
-                if not model.col_fit_deriv:
-                    return [np.ravel(_) for _ in
-                            (np.ravel(weights) * np.array(model.fit_deriv(x, y, *params)).T).T]
-                return [np.ravel(_) for _ in weights * np.array(model.fit_deriv(x, y, *params))]
+class TRFLSQFitter(_NLLSQFitter):
+    """
+    Trust Region Reflective algorithm and least squares statistic.
+
+    Parameters
+    ----------
+    calc_uncertainties : bool
+        If the covarience matrix should be computed and set in the fit_info.
+        Default: False
+    check_bounds : bool
+        If the set parameter bounds for a model will be enforced each given
+        parameter while fitting via a simple min/max condition. A True setting
+        will replicate how LevMarLSQFitter enforces bounds.
```

Comment:
```suggestion
        will replicate how TRFLSQFitter enforces bounds.
```
?

Reply:
This suggestion made me realize that the term `check_bounds` was misleading because this suggestion would change the meaning to the exact opposite of what the setting is being used for. Thus in 2281f84, I changed the term from `check_bounds` to `use_min_max_bounds`.

For those interested the `use_min_max_bounds` setting is used to indicate if the bounds enforcement algorithm should use the simple min/max (after each iteration the value of a parameter is set to the max of min bound and current value and then to the min of the max bound and current value), which is how the `LevMarLSQFitter` enforces parameter bounds. The TRF (and dogbox) algorithm can implicitly enforce a passed bound during fitting, this eliminates possible issues with fitting that arise from fits that locally (during intermediate iterations) tend near a bound from sticking to a bound value.